### PR TITLE
Add methods to Shape

### PR DESCRIPTION
**Pull request by topecongiro**
While reading the code, I found following two patterns to be pervasive.

```rust
// 1
let max_width = context.config.max_width - indent.width();
let shape = Shape::legacy(max_width, indent);
```
```rust
// 2
let new_shape = Shape {
    width: context.config.max_width - shape.indent.width(),
    **shape
}
```

This PR adds above two methods and does minor refactorings.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -254,6 +246,21 @@ impl Shape {
         }
     }
 
+    pub fn indented(config: &Config, indent: Indent) -> Shape {
```
Comment (nrc) [inline src/lib.rs]: Could you swap the arg order please, I think that would be more consistent with other functions

File: src/lib.rs [hunk 2]
```diff
@@ -254,6 +246,21 @@ impl Shape {
...
+        Shape::legacy(config.max_width.checked_sub(indent.width()).unwrap_or(0),
+                      indent)
```
Comment (nrc) [inline src/lib.rs]: Do we save much code by using the `legacy` ctor vs init'ing `Shape` directly? I'd prefer the latter unless there is a big saving in code complexity

**Comment by topecongiro** [comment]
Thanks for comments! Rebased.

**Comment by nrc** [comment]
> Thanks for comments! Rebased.

I don't see any changes, did you forget to push?

**Comment by nrc** [comment]
@topecongiro could you email please - I can't find any contact info for you? Use my GitHub username at mozilla.com

**Comment by topecongiro** [comment]
Oops, I am sorry.

**Comment by nrc** [comment]
> Oops, I am sorry.

No worries, and thanks for the changes

**Pull request merged**